lib/rubocop/ext/regexp_parser.rb

```diff
@@ -68,7 +68,9 @@ def build_location
             return { body: expression } unless (q = quantifier)
 
             body = expression.adjust(end_pos: -q.text.length)
-            q_loc = expression.with(begin_pos: body.end_pos)
+            q.origin = origin
```

Comment:
these changes fix a bug:

as i noticed in [this test](https://github.com/rubocop/rubocop/pull/11565/files#diff-2955709183e371bdcab246c6a457bf346eb5ed3dbcd6626bf2db521f1163ae9dR124), the quantifier location was previously incorrect if the regexp was in x-mode and there was whitespace between the quantified expression and the quantifier.

e.g. `/a   +/x`.

in such cases, the starting position of the quantifier is *not* the end position of the body.

as far as i could tell, this quantifier location was not used by any existing cop so far, so i though this bugfix was not worth its own PR.